src/integer/poly_over_z/from.rs

```diff
@@ -52,18 +52,30 @@ impl FromStr for PolyOverZ {
     /// [`InvalidStringToCStringInput`](MathError::InvalidStringToCStringInput)
     /// if the provided string contains a Null Byte.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // remove whitespaces at the start and at the end
+        let s_trimmed = s.trim();
+
+        if s_trimmed == "0" {
+            return Ok(Self::default());
+        }
+
+        // fmpz_poly_set_str just skips the two symbols after the number of
+        // coefficients (even if they are not whitespaces), hence we have to check if
+        // they are whitespaces manually.
+        // and we only have to check it once, because for every other position it checks
+        // whether there is only one whitespace
```

Comment:
It ensures that the first character behind the first number is a space.
Interesting Sitenote: whitespaces that are not spaces can be included within numbers, e.g. `\n`, `\t` and will be ignored.

```rust
let a = PolyOverZ::from_str("2  1\n2 3").unwrap();
let b = PolyOverZ::from_str("2  12 3").unwrap();
assert_eq!(a,b);
```

```suggestion
        // fmpz_poly_set_str just skips the two symbols after the first space
        // behind the number of coefficients (even if not a space), hence
        // it has to be checked here to ensure that no number is lost.
        // We only have to check it once, because for every other position it checks
        // whether there is only one space.
```

Reply:
interesting, I added your suggestion, do you want me to also include this sitenote?

Reply:
No, just the suggestion is enough. The other thing is just something interesting to know that I stumbled over during the review.